docs/wiki/Cryptography/post-quantum-cryptography.md

```diff
@@ -0,0 +1,36 @@
+# Post-Quantum Cryptography
+
+Classical cryptography safeguards information by leveraging the inherent difficulty of certain mathematical problems. These problems fall under the area of mathematical research called the ["Hidden Subgroup Problem (HSP)"](https://en.wikipedia.org/wiki/Hidden_subgroup_problem). Imagine a large group with a secret subgroup known only to insiders, these problems makes determining the structure of the secret subgroup (size, elements) computationally intractable for an outsider. Whereas, someone with the "secret" (the private key) can easily identify the subgroup.
+
+Public-key cryptography leverages this concept. Algorithms like RSA, DSA, and [ECDSA](/wiki/Cryptography/ecdsa.md) rely on hidden subgroup problems like prime factorization of large integers or discrete logarithm calculations to secure private keys. The difficulty of solving these problems increases exponentially with key size, making brute-force attacks impractical for classical computers. This inherent difficulty safeguards encrypted data.
+
+However, the landscape is shifting.
+
+Quantum computers, harnessing the principles of quantum mechanics, offer novel computational approaches. Certain quantum algorithms can solve these classical cryptographic problems with exponential efficiency compared to their classical counterparts. This newfound capability poses a significant threat to the security of data encrypted with classical cryptography.
```

Comment:
I would elaborate on this part a little bit. Maybe mentioning qubits and that current quantum computers don't have enough of them to break crypto but there are threats like HNDL https://wikipedia.org/wiki/Harvest_now,_decrypt_later.